check if the input clue is an alphabetic word

rsun45 opened the pull request:
<img width="584" alt="截屏2021-05-05 上午11 02 06" src="https://user-images.githubusercontent.com/22154220/117169316-01651200-ad97-11eb-9fc2-bffda7eafac6.png">
<img width="583" alt="截屏2021-05-05 上午11 02 43" src="https://user-images.githubusercontent.com/22154220/117169358-09bd4d00-ad97-11eb-9a9f-491edb5a6c3d.png">
<img width="590" alt="截屏2021-05-05 上午11 02 17" src="https://user-images.githubusercontent.com/22154220/117169384-0f1a9780-ad97-11eb-854b-21a10a24ce05.png">


Player can type until valid word is inputed

rsun45 committed:
check if the input clue is an alphabetic word

Player can type until valid word is inputed

bonnieli commented on client/src/components/ClueModal.js:
```diff
@@ -53,9 +56,20 @@ const ClueModal = (props) => {
         setNumOfGuesses(e.target.value);
     }
 
+    const isOneAlphabeticWord = (word) => {
```
Checking if the word is not one of the clue words

rsun45 replied:
This is implemented.

rsun45 committed:
checking if entered clue word is same with card words

Now player cannot set card words as clue, no matter upper or lower cases or spaces are used.

rsun45 commented:
<img width="327" alt="截屏2021-05-05 下午8 02 08" src="https://user-images.githubusercontent.com/22154220/117224125-32683580-addd-11eb-9075-d373ed10bf57.png">
<img width="322" alt="截屏2021-05-05 下午8 02 39" src="https://user-images.githubusercontent.com/22154220/117224127-3300cc00-addd-11eb-99e4-3f381b94c91b.png">

bonnieli commented on client/src/components/ClueModal.js:
```diff
@@ -53,9 +56,38 @@ const ClueModal = (props) => {
+    const isCardWord = (word) => {
+        const cards = props.cards;
+        let i, j;
+        for (i = 0; i < cards.length; i++) {
```
this can be rewritten so that it's not so complicated 

1. The incoming parameter `word` can be trimmed at the beginning and saved as a constant and transformed to lowercase
2. If you're looping through all the cards, you can do something like the following in a couple of lines

```
const findCardWord = cards[0].find(card => card.word.toLowerCase() === word)
```

I'm not to sure what `props.cards` looks like, if it's an array of just one element, bceause your second nested for loop only ever accesses the first element of the array

bonnieli commented on client/src/components/ClueModal.js:
```diff
@@ -53,9 +56,38 @@ const ClueModal = (props) => {
+        const cards = props.cards;
+        let i, j;
+        for (i = 0; i < cards.length; i++) {
+          for (j = 0; j < cards[0].length; j++) {
```
```suggestion
          for (let j = 0; j < cards[0].length; j++) {
```

rsun45 committed:
optimized code

rsun45 committed:
Merge branch 'dev' into Limit-one-word-clue-word-check
